modules/ding_adgangsplatformen/ding_adgangsplatformen.module

```diff
@@ -335,6 +370,25 @@ function ding_adgangsplatformen_callback() {
       $user_info = $response->getBody()->getContents();
       $user_info = drupal_json_decode($user_info);
 
+      $conf = ding_adgangsplatformen_get_configuration();
+      if ($conf['tokenClientId'] && $conf['tokenClientSecret']) {
+        if (!empty($user_info['attributes']['userId']) && !empty($user_info['attributes']['pincode'])) {
+          // Get a token with less privileges by logging in with name and pin on
+          // another client.
+          $token_provider = ding_adgangsplatformen_get_provider();
+          $token_access_token = $token_provider->getAccessToken('password', [
+                                  'username' => $user_info['attributes']['userId'],
+                                  'password' => $user_info['attributes']['pincode'],
+                                ]);
+
+          $_SESSION['ding_adgangsplatformen_openplatform_token_get'] = $token_access_token->getToken();
```

Comment:
There is a lot of reaching into the session throughout the code. How about implementing a setter corresponding to `ding_adgangsplatformen_openplatform_token_get` - or otherwhise change the function to `ding_adgangsplatformen_openplatform_token` and allow it both set and get the value.